src/Model.php

```diff
@@ -600,6 +661,32 @@ public function isDirty($fields = [])
         return false;
     }
 
+    /**
+     * @param string|null $filter
+     *
+     * @return array
+     */
+    public function getFields(string $filter = null)
+    {
+        if (!$filter) {
+            return $this->fields;
+        }
+
+        return array_filter($this->fields, function ($field, $name) use ($filter) {
+
+            // do not return fields outside of "only_fields" scope
+            if ($this->only_fields && !in_array($name, $this->only_fields)) {
+                return false;
+            }
+
+            switch ($filter) {
+                case 'system': return $field->system;
+                case 'editable': return $field->ui['editable'] ?? !$field->system;
```

Comment:
Should probably use $field->isEditable() and $field->isVisible() methods here because:
1. DRY principle
2. editable will then take read_only and never_persist properties into account too.

Reply:
ok, changed, also dropped check for system in this case. Test continue to work, but i haven't done any tests for read-only and never_persist.